### PR TITLE
Build Pipeline Refactoring - Phase 1

### PR DESCRIPTION
**Pull request by joshedney**
## Goal

Refactor the build pipeline to fall in line with other repositories. This is phase 1.

## Changeset

- Add conditional trigger to run the full pipeline when `[full ci]` is in the commit message.
- Build 4.23 and 4.27 plugins on a normal build and build/run test fixtures for 4.27
- Build all other (4.25/4.26/5.0EA) on a full pipeline run.

## Testing

Normal Build - https://buildkite.com/bugsnag/bugsnag-unreal/builds/842
Full CI Build - https://buildkite.com/bugsnag/bugsnag-unreal/builds/843

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -100,13 +49,12 @@ steps:
...
-  ##############################################################################
   #
-  # Build E2E Fixtures
+  # Build Test Fixtures
   #
-  ##############################################################################
 
-  - name: 'Build Android test fixtures - UE 4.27'
+  # UE 4.27
+  - name: ':android: Build Test Fixtures - UE 4.27'
```
Comment (twometresteve) [inline .buildkite/pipeline.yml:57]: Personally I like having the platform spelled out in the step name, not just using the iOS/macOS/Android emoji.  I find it easier to read and also allows you to "Find on page" in the browser should you so wish.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -163,25 +111,24 @@ steps:
...
+  # E2E Tests
   #
-  ##############################################################################
 
-  - label: 'E2E Tests - Android 11'
+  # UE 4.27
+  - label: 'E2E Tests - Android 11 - UE 4.27'
     depends_on:
-      - android_fixture
+      - android_fixture_4_27
```
Comment (twometresteve) [inline .buildkite/pipeline.yml:123]: The inconsistency with other steps makes me a little twitchy.  I'd make all of them look like:
```
  depends_on: some_step
```
(and without the `[]`s on other steps).

**Review by twometresteve — COMMENTED** [review]
I like this setup because:
- It's really clear what's in a basic and full build
- There are only two pipeline files to think about
- There is only one other file to consider, containing a single `if`.
- It's consistent with Unity and where other repos are now likely to head

But let's also bring @nickdowell in; we can normally trust him to have an opinion.

**Review by twometresteve — COMMENTED** [review]
I'm happy if Nick's happy.

**Review by nickdowell — APPROVED** [review]
👍

Would be nice to use GitHub's "Squash and merge" option to keep the history clean - we don't need to see all those "[full ci]" commits forever 🙃

**Pull request merged**